test/System.Linq.Dynamic.Core.Tests/Util/ThreadSafeSlidingCacheTests.cs

```
@@ -0,0 +1,152 @@
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Linq.Dynamic.Core.Tests.Util
+{
+    public class ThreadSafeSlidingCacheTests
+    {
+        [Fact]
+        public void ThreadSafeSlidingCache_CacheOperations()
+        {
+            var mockDateTime = new MockDateTimeProvider();
+
+            // Arrange
+            var cache = new ThreadSafeSlidingCache<int, string>(
+                TimeSpan.FromSeconds(1), 
+                dateTimeProvider: mockDateTime);
+
+            // Add
+            cache.AddOrUpdate(1, "one");
+            cache.AddOrUpdate(2, "two");
+            cache.AddOrUpdate(3, "three");
+
+            // Replace
+            cache.AddOrUpdate(1, "oneone");
+
+            Assert.True(cache.Count == 3, $"Expected 3 items in the cache, only had {cache.Count}");
+
+
+            // Test retrieval
+            Assert.True(cache.TryGetValue(1, out var value1), $"Expected to find the value, but did not");
+            Assert.True(cache.TryGetValue(2, out var value2), $"Expected to find the value, but did not");
+            Assert.True(cache.TryGetValue(3, out var value3), $"Expected to find the value, but did not");
+
+            // Test Removal
+            cache.Remove(1);
+            Assert.True(cache.Count == 2, $"Expected 2 items in the cache, only had {cache.Count}");
+
+        }
+
+        [Fact]
+        public void ThreadSafeSlidingCache_TestExpire()
+        {
+            var mockDateTime = new MockDateTimeProvider();
+
+            // Arrange
+            var cache = new ThreadSafeSlidingCache<int, string>(TimeSpan.FromMinutes(10),
+                dateTimeProvider: mockDateTime);
+
+            // Act
+            cache.AddOrUpdate(1, "one");
+            mockDateTime.UtcNow = mockDateTime.UtcNow.AddMinutes(11);
+            if (cache.TryGetValue(1, out var value))
+            {
+                Assert.True(false, $"Expected to not find the value, but found {value}");
```

In this case, use `Asser.Fail`

---

Lol, that's a Co-pilot fail!  Fixed! How did I miss that!

---

So, going back to this - having trouble with xUnit - does not appear they have `Assert.Fail(..)` - I think Co-pilot suggested that as an alternative.  Some StackOverflow articles show I should just throw an exception instead, which will fail it. I am not sure how I feel about that.  How would you want me to trigger a failure?

---

I understand. I was using an older version from xunit in the solution.
It's added here: https://github.com/xunit/xunit/issues/2609#issuecomment-1399639884

If you want; you upgrade all unit-test projects to use 2.5.0 which includes Assert.Fail

Else if this too much work, just keep the existing code.